invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/oaipmh/details/LinksTable.js

```diff
@@ -5,53 +5,166 @@
+      try {
+        const response = await http.get(
+          '/api/oaipmh/formats'
+        );
+        const formats = response.data?.hits?.hits;
+        if (Array.isArray(formats) && formats.length > 0) {
+          const serialized = formats.map((formt) => {
+            return {
+              key: formt.id,
+              value: formt.id,
+              text: knownFormats[formt.id] ?? formatKeyToName(formt.id)
+            }
+          });
+          setFormats(serialized);
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    }
+    getFormats();
+
+  }, []);
+
   /**
-   * WIP: this method should remove metadataPrefix from the url once the attribute is listed
-   * as a dropdown.
+   * Replaces the metadata prefix in the link.
+   * Used to update links when the metadata prefix is changed.
+   */
+  const replaceLinkPrefix = (link, newPrefix) => {
+    const oldPrefix = getPrefixFromLink(link);
+    if (oldPrefix) {
+      return link.replace(oldPrefix, newPrefix)
+    }
+    return link;
+  };
+
+  /**
+   * Retrieves metadata prefix from a link.
+   * Uses a regex to retrieve the prefix, matching "&metadataPrefix=...&".
    *
-   * @param {string} url
-   * @returns {string} stripped url
+   * Returns null if pattern not found.
+   */
+  const getPrefixFromLink = (link) => {
+    const prefixRegex = /\&metadataPrefix\=(.*)\&/;
+    const match = link.match(prefixRegex);
+    if (match && match.length) {
+      const prefix = match[1];
+      return prefix;
+    }
+    return null;
+  };
+
+  /**
+   * Transforms a key into a readable name.
+   * @example
+   *  // returns Oai Datacite
+   *  formateKeytoName('oai_datacite')
    */
-  const sanitizeLink = (url) => {
-    return url;
+  const formatKeyToName = (formatKey) => {
+    const whiteSpaced = formatKey.replace("_", " ");
+    const capitalised = capitalize(whiteSpaced);
+    return capitalised;
   };
 
-  const { links } = data;
-  const listRecords = sanitizeLink(links["oai-listrecords"]);
-  const listIdentifiers = sanitizeLink(links["oai-listidentifiers"]);
+  /**
+   * Dropdown's on change handler.
+   * This method will update the following states if the new prefix is known:
+   * - 'links', each link with its prefix replaced by the new prefix.
+   */
+  const prefixOnChange = (event, data) => {
+    const newPrefix = data.value;
+    if (formats.some(obj => obj.key === newPrefix)) {
+      const newLinks = {};
+      Object.keys(links).map((key) => {
+        let link = links[key];
+        let newLink = replaceLinkPrefix(link, newPrefix);
+        newLinks[key] = newLink;
+      });
+      setLinks(newLinks);
+    }
+  };
+
+  const listRecords = links["oai-listrecords"];
+  const listIdentifiers = links["oai-listidentifiers"];
+
+  // Set default prefix based on backend link, use dublin core if unknown.
+  let defaultPrefix = getPrefixFromLink(listRecords)
+  if (!Object.prototype.hasOwnProperty.call(formats, defaultPrefix)) {
```

Comment:
Suggestion: for more complex code, prefer assigning expressions to a `const` with easy-to-understand names.
```suggestion
  const hasDefaultPrefix = Object.prototype.hasOwnProperty.call(formats, defaultPrefix);
  if (!hasDefaultPrefix) {
```